javascript/permalink.js

```diff
@@ -1,119 +1,52 @@
-                control.ignoredLayers = this.ignoredLayers;
-                break;
-            }
-        }
-    },
-    createParams: function(center, zoom, layers) {
-        center = center || this.map.getCenter();
-
-        var params = OpenLayers.Util.getParameters(this.base);
-
-        if (center) {
-            // Call parent method
-            params = OpenLayers.Control.Permalink.prototype.createParams.apply(this, [center, zoom, layers]);
-
-            // Override layers param
-            params.layers = this.getLayerString(layers);
-        }
-
-        return params;
-    },
-    getLayerString: function(layers) {
-        var layers = layers || this.map.layers;
-        var result = '';
-        for (var i = 0, len = layers.length; i < len; i++) {
-            var layer = layers[i];
-            var id = layer.layerId || 0;
-            var flag;
-            if (id < 1) {
-                // No layerId set -> ignore the layer
-                continue;
-            }
-            if (layer.isBaseLayer) {
-                flag = (layer == this.map.baseLayer) ? 'B' : '0';
-            } else {
-                flag = (layer.getVisibility()) ? 'T' : 'F';
-            }
-            result = this.setFlag(result, flag, id);
-        }
-        return result;
-    },
-    setFlag: function(config, flagValue, id) {
-        // The layers can be in different order
-        while (config.length < id) {
-            // Fill with defaults
-            config = config + 'F';
-        }
-        return config.substr(0, id-1) + flagValue + config.substr(id);
+// Create a url from the current map's view.
+function getPermalink(otherParams = {}) {
+  const zoom = parseInt(map.getView().getZoom(), 10);
+  const [lon, lat] = ol.proj.toLonLat(map.getView().getCenter());
+
+  const mapLayers = map.getLayers().getArray();
+  const maxLayerId = mapLayers.reduce(
+    (maxLayerId, layer) => Math.max(maxLayerId, layer.get("layerId")),
+    0
+  );
+  let layers = "";
+  for (var i = 1; i <= maxLayerId; i += 1) {
+    const layer = mapLayers.find((l) => l.get("layerId") === i);
+    if (!layer?.getVisible()) {
+      layers += "F";
+    } else if (layer.getVisible()) {
+      layers += layer.get("isBaseLayer") ? "B" : "T";
     }
-});
+  }
+
+  const params = new URLSearchParams({
+    zoom,
+    lon: lon.toFixed(5),
+    lat: lat.toFixed(5),
+    layers,
+    ...otherParams,
+  });
+
+  return window.location.href.split("?")[0] + `?${params.toString()}`;
+}
+
+class Permalink extends ol.control.Control {
+  constructor() {
+    super({
+      element: document.createElement("div"),
+    });
+    this.timeout = null;
+
+    this.aElement_ = document.createElement("a");
+    this.aElement_.innerHTML = "Permalink";
+    this.element.className = "ol-permalink";
+    this.element.appendChild(this.aElement_);
+  }
+
+  render() {
+    // We use a timeout to avoid unnecessary updates.
```

Comment:
The way I understand the code is that it does not really remove the updates but moves them past the phase where the map would be scrolling. It would not actually cancel redundant or outdated updates.
To be clear, I don't see this as a problem, just trying to understand what this does.

Reply:
The timeout makes the HTML updates  wait that  the map is in idle state for 500ms then update the permalink with the last state. It just avoid having the html rerendered dozen times instead of one.